src/test/java/test/listeners/invokeasinsertionorder/ListenerInvocationAsInsertOrderTest.java

```diff
@@ -0,0 +1,140 @@
+package test.listeners.invokeasinsertionorder;
+
+import static test.listeners.invokeasinsertionorder.Constants.*;
+import java.util.Arrays;
+import java.util.List;
+import org.assertj.core.api.Assertions;
+import org.testng.ITestNGListener;
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import test.SimpleBaseTest;
+
+public class ListenerInvocationAsInsertOrderTest extends SimpleBaseTest {
+    
+  @Test
+  public void test() {
```

Comment:
Am guessing that this class will be required to add more tests such as listener addition for a JUnit based test or listeners that are added via other means etc.,  So it would be great if the method name can be renamed to something which is more apt than being generic.

Reply:
Yes, I will update the name.